autoswitch_virtualenv.plugin.zsh

```diff
@@ -38,10 +38,18 @@ function _check_venv()
 
         SWITCH_TO=""
 
+        # Get the .venv file, scanning parent directories
         venv_path=$(_check_venv_path "$PWD")
         if [[ -n "$venv_path" ]]; then
-          file_owner="$(stat -c %u "$venv_path")"
-          file_permissions="$(stat -c %a "$venv_path")"
+
+          stat --version &> /dev/null
+          if [[ $? -eq 0 ]]; then   # Linux, or GNU stat
```

Comment:
Sure, they would work just because it is based on the `stat` binary that is actually working.